lib/happo/public/HappoDiffs.jsx

```diff
@@ -94,21 +100,91 @@ HappoNewImages.propTypes = {
   images: PropTypes.arrayOf(PropTypes.shape(imageShape)),
 };
 
+const Swiper = React.createClass({
+  propTypes: {
+    previous: PropTypes.string.isRequired,
+    current: PropTypes.string.isRequired,
+  },
+
+  getInitialState() {
+    return {
+      cursorLeftPercent: 50, // Start in the center
+    };
+  },
+
+  _handleMouseMove(event) {
+    const leftPx = event.pageX - event.target.offsetLeft;
+    const leftPercent = leftPx / event.target.offsetWidth * 100;
+
+    console.log(event.target)
+    console.log(leftPx, event.target.offsetWidth, leftPercent);
+    this.setState({
+      cursorLeftPercent: leftPercent,
+    })
+  },
+
+  render() {
+    const {
+      previous,
+      current,
+    } = this.props;
+
+    const { cursorLeftPercent } = this.state;
+    const clipPath =
+      `polygon(${cursorLeftPercent}% 0, 100% 0, 100% 100%, ${cursorLeftPercent}% 100%)`;
+
+    return (
+      <div
```

Comment:
I don't see it as a hard rule, but I generally find that this pattern makes an app more understandable and testable. It's kinda like what Dan outlines here: https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0